junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGeneration.java

```diff
@@ -0,0 +1,168 @@
+
+	Style value() default Style.DEFAULT;
+
+	enum Style implements DisplayNameGenerator {
+		/**
+		 * Default display name generator.
+		 */
+		DEFAULT {
+			/**
+			 * TODO Javadoc
+			 */
+			@Override
+			public String generateDisplayNameForClass(Class<?> testClass) {
+				Preconditions.notNull(testClass, "Test class must not be null");
+				String name = testClass.getName();
+				int index = name.lastIndexOf('.');
+				return name.substring(index + 1);
+			}
+
+			/**
+			 * TODO Javadoc
+			 */
+			@Override
+			public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+				Preconditions.notNull(nestedClass, "Nested test class must not be null");
+				return nestedClass.getSimpleName();
+			}
+
+			/**
+			 * TODO Javadoc
+			 */
+			@Override
+			public String generateDisplayNameForMethod(Method testMethod) {
+				Preconditions.notNull(testMethod, "Test method must not be null");
+				return String.format("%s(%s)", testMethod.getName(),
+					ClassUtils.nullSafeToString(Class::getSimpleName, testMethod.getParameterTypes()));
+			}
+		},
+
+		/**
+		 * TODO Javadoc
+		 */
+		UNDERSCORE {
+			@Override
+			public String generateDisplayNameForClass(Class<?> testClass) {
+				return replaceUnderscore(DEFAULT.generateDisplayNameForClass(testClass));
+			}
+
+			@Override
+			public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+				return replaceUnderscore(DEFAULT.generateDisplayNameForNestedClass(nestedClass));
+			}
+
+			@Override
+			public String generateDisplayNameForMethod(Method testMethod) {
+				return replaceUnderscore(DEFAULT.generateDisplayNameForMethod(testMethod));
+			}
+
+			private String replaceUnderscore(String name) {
+				return name.replace('_', ' ');
+			}
+		},
+
+		/**
+		 * TODO Javadoc
+		 */
+		CAMEL_CASE {
+			private final Pattern camelCasePattern = Pattern.compile(String.format("%s|%s|%s",
+				"(?<=[A-Z])(?=[A-Z][a-z])", "(?<=[^A-Z])(?=[A-Z])", "(?<=[A-Za-z])(?=[^A-Za-z])"));
+
+			@Override
+			public String generateDisplayNameForClass(Class<?> testClass) {
+				return splitCamelCase(DEFAULT.generateDisplayNameForClass(testClass));
+			}
+
+			@Override
+			public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+				return splitCamelCase(DEFAULT.generateDisplayNameForNestedClass(nestedClass));
+			}
+
+			@Override
+			public String generateDisplayNameForMethod(Method testMethod) {
+				return splitCamelCase(DEFAULT.generateDisplayNameForMethod(testMethod));
+			}
+
+			private String splitCamelCase(String name) {
+				return camelCasePattern.matcher(name).replaceAll(" ");
+			}
+		},
+
+		/**
+		 * TODO Javadoc
+		 */
+		SENTENCES {
+			@Override
+			public String generateDisplayNameForClass(Class<?> testClass) {
+				return DEFAULT.generateDisplayNameForClass(testClass);
+			}
+
+			@Override
+			public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+				return nestedClass.getSimpleName().replace('_', ' ') + "...";
```

Comment:
I don't understand why an ellipsis would be desired at this level but not for the enclosing, top-level class.

Reply:
The name of the test class is used as the caption, the title, the specification. You could prepend it to each sentence like:
- "Feature A Spec: B is given [...] when doing X [...] then [...]."
- "Feature A Spec: C is given [...] when doing Y [...] then [...]."
- _"Feature A Spec: {P} Q {R}"_

So, you usually don't want to repeat the top-most enclosing box over and over again. The really intersting part is the start of each sentence, which start at level 1 (when 0 is the top-level).

Reply:
Start watching here https://vimeo.com/289852238#t=1370s for minute or two. Or five. :)

https://vimeo.com/289852238#t=1370s